File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpHarvestToolData.java
package calebxzhou.rdi.mc.common2.mcp;

import java.util.List;
import java.util.Map;

public record RMcpHarvestToolData(
        String stateSource,
        Block block,
        List<ToolScenario> scenarios,
        List<String> notes
) {
    public record Block(
            String id,
            String state,
            RMcpBlockPosData pos,
            boolean requiresCorrectToolForDrops,
            List<String> mineableWith,
            String minimumTier
    ) {
    }

    public record Tool(String id, String category, String tier, Map<String, Integer> enchantments) {
    }

    public record Drop(String id, int countMin, int countMax, String snbt) {
    }

    public record ToolScenario(
            Tool tool,
            boolean correctToolForDrops,
            boolean harvestable,
            List<Drop> drops
    ) {
    }
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpHarvestToolData.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpHarvestToolData(
    val stateSource: String,
    val block: Block,
    val scenarios: List<ToolScenario>,
    val notes: List<String>
) {
    @JvmRecord
    data class Block(
        val id: String,
        val state: String,
        val pos: RBlockPos,
        val requiresCorrectToolForDrops: Boolean,
        val mineableWith: List<String>,
        val minimumTier: String?
    )

    @JvmRecord
    data class Tool(
        val id: String,
        val category: String,
        val tier: String,
        val enchantments: Map<String, Int>?
    )

    @JvmRecord
    data class Drop(val id: String, val countMin: Int, val countMax: Int, val snbt: String?)

    @JvmRecord
    data class ToolScenario(
        val tool: Tool,
        val correctToolForDrops: Boolean,
        val harvestable: Boolean,
        val drops: List<Drop>
    )
}
